Alerting: Fix simplified routing group by override

JacobsonMT opened the pull request:
**What is this feature?**

Custom group by overrides for simplified routing looked like they were generating correctly using `GroupByStr`, displaying correctly in exports and UI but the generated routes were actually missing required fields normally set during upstream Route [validation](https://github.com/grafana/alerting/blob/bc622368256dbf116357d232441b6d1ad6166363/definition/alertmanager_validation.go#L17-L23).

This fix ensures the missing fields `GroupBy` and `GroupByAll` are applied to the generated routes.

**Why do we need this feature?**

For simplified routing group by overrides to function correctly.

**Who is this feature for?**

Users of simplified routing.

yuri-tceretian commented on pkg/services/ngalert/notifier/autogen_alertmanager.go:
```diff
@@ -77,6 +77,10 @@ func newAutogeneratedRoute[R receiver](ctx context.Context, logger log.Logger, s
+		if skipInvalid {
+			logger.Error("Generation of simplified routing policies failed. Skipping", "error", err)
+			return newAutogenRoute, nil
+		}
```
it looks it will fail all autogen routes if at least one is not valid. It would be great if we could skip on 1 level down - contact point level

JacobsonMT committed:
Alerting: Fix simplified routing custom group by override

Custom group by overrides for simplified routing were missing required fields
GroupBy and GroupByAll normally set during upstream Route validation.

This fix ensures those missing fields are applied to the generated routes.

yuri-tceretian reviewed (approved):
LGTM

JacobsonMT committed:
Inline GroupBy and GroupByAll initialization instead of normalize after
